packages/ui/client/components/dashboard/TestsEntry.cy.tsx

```diff
@@ -0,0 +1,30 @@
+import TestsEntry from './TestsEntry.vue'
+
+const passEntrySelector = '[data-testid=pass-entry]'
```

Comment:
Shouldn't we share and export that testids from components?

Reply:
I've never seen a pattern like that in a codebase, nor have I implemented it before because:

1. The indirection makes debugging why a certain element wasn't found more difficult.

Instead of one step: Cmd+F `data-testid="pass-entry-selector"`

You'd need to do two: Cmd+F `pass-entry-selector` and then Cmd+F `passEntrySelector` to find its usage within the template.

2. The boilerplate required makes the impl less readable.
3. DataTestID is only used some of the time for selectors. `[role="button"]` and other selector strategies are also valid, and you'd create an inconsistency

For the implementation, I guess you would do:

```vue
<script lang="ts">
export const passEntryDataTestId = 'pass-entry-selector'
</script>

<script setup lang="ts">
// ....
</script>

<template>
	<div :data-testid="passEntryDataTestId" />
</template>
```

```js
import Component, { passEntryDataTestId } from './Component.vue'
const passEntrySelector = `[data-testid=${passEntryDataTestId}]`
// test...
```

I would never recommend doing this. I hope this helps answer your question.

Reply:
One of the patterns that isn't as common nowadays than it was for QA Engineers is the Page Object Model pattern. This pattern essentially has you build a Class representation of the "Pages" in your application. It was very common pre-2015 when writing Selenium tests.

Cypress doesn't really have good patterns for PageObjectModel (We actually discourage users from doing it, though I've never really agreed with why.) I've considered exploring this pattern again from the perspective of developers instead of QA Engineers, but haven't done so.